vault/resource_ldap_secret_backend.go

```diff
@@ -0,0 +1,285 @@
+			Type:        schema.TypeString,
+			Optional:    true,
+			Sensitive:   true,
+			Description: "Client certificate key to provide to the LDAP server, must be x509 PEM encoded.",
+		},
+		consts.FieldInsecureTLS: {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Computed:    true,
+			Description: "Skip LDAP server SSL Certificate verification - insecure and not recommended for production use.",
+		},
+		consts.FieldLength: {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Deprecated:  "Length is deprecated and password_policy should be used with Vault >= 1.5.",
+			Description: "The desired length of passwords that Vault generates.",
+		},
+		consts.FieldPasswordPolicy: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Name of the password policy to use to generate passwords.",
+		},
+		consts.FieldSchema: {
+			Type:        schema.TypeString,
+			Default:     "openldap",
+			Optional:    true,
+			Description: "The LDAP schema to use when storing entry passwords. Valid schemas include openldap, ad, and racf.",
+		},
+		consts.FieldRequestTimeout: {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Default:     90,
+			Description: "Timeout, in seconds, for the connection when making requests against the server before returning back an error.",
+		},
+		consts.FieldStartTLS: {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Computed:    true,
+			Description: "Issue a StartTLS command after establishing unencrypted connection.",
+		},
+		consts.FieldUPNDomain: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			Description: "Enables userPrincipalDomain login with [username]@UPNDomain.",
+		},
+		consts.FieldURL: {
+			Type:        schema.TypeString,
+			Default:     "ldap://127.0.0.1",
+			Optional:    true,
+			Description: "LDAP URL to connect to (default: ldap://127.0.0.1). Multiple URLs can be specified by concatenating them with commas; they will be tried in-order.",
+		},
+		consts.FieldUserAttr: {
+			Type:        schema.TypeString,
+			Default:     "cn",
+			Optional:    true,
+			Description: "Attribute used for users (default: cn)",
+		},
+		consts.FieldUserDN: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "LDAP domain to use for users (eg: ou=People,dc=example,dc=org)",
+		},
+	}
+	resource := provider.MustAddMountMigrationSchema(&schema.Resource{
+		CreateContext: createLDAPConfigResource,
+		UpdateContext: createLDAPConfigResource,
+		ReadContext:   ReadContextWrapper(readLDAPConfigResource),
+		DeleteContext: deleteLDAPConfigResource,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		CustomizeDiff: getMountCustomizeDiffFunc(consts.FieldMount),
+		Schema:        fields,
+	})
+
+	// Add common mount schema to the resource
+	provider.MustAddSchema(resource, getMountSchema("path", "type"))
```

Comment:
Related to comment above; we can avoid excluding `path` here if we decide to go this route over `mount`